app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -237,6 +237,37 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
         var ga4LinkData = JSON.parse(module.getAttribute('data-ga4-link'))
         ga4LinkData.index_total = totalLinks
         module.setAttribute('data-ga4-link', JSON.stringify(ga4LinkData))
+      },
+
+      // index is given as a string of the form 1.2.3 or 1.2
+      // split this into named sub-parameters
+      createIndexObject: function (index) {
+        if (typeof index === 'undefined') {
+          return undefined
+        }
+        if (typeof index === 'object') {
+          return index
+        }
+
+        index = index.toString().split('.')
+        // this will soon include 'index_section_count'
+        // number of sections overall in the thing being tracked (but not yet in use)
+        switch (index.length) {
+          case 1:
+            return {
+              index_section: parseInt(index[0])
```

Comment:
Quick question: I remember us saying that the `index` attribute on accordions would only ever contain one digit and that it would need to be mapped to `index_section`, which is what the above code would do. I'm just wondering about what would happen with links that only have one digit, like the breadcrumbs; would mapping the breadcrumb indexes to `index_section` be okay? Or would it need to be `index_link`?

Reply:
That's concerning - if the indexes are that inconsistent this may prevent us from doing this entire approach - we'll have to manually update all the existing indexes in every component and component call across all our applications. Which I was hoping to avoid. Will raise.

Reply:
The Breadcrumbs GA4 stuff is only defined in one place, so maybe it could have the correct object added straight into it's `ga4-link` attribute? Or is that complicating things because then we'll have `1.2.X` in some components and `index: {index_link: 1}` in others

Reply:
I've written the code to be future proof - if the parameter passed to the function is already an object, it'll simply return the object, so for those things that are an edge case we could update them manually. Good shout.

Reply:
Cool, sounds like we've got a solution (manually updating the edge cases) so I'll mark this as resolved 👍